inc/class-purge-cache.php

```diff
@@ -0,0 +1,183 @@
+		if ( 'publish' !== $new_status && 'publish' !== $old_status ) {
+			return;
+		}
+
+		// Purge cache.
+		$this->fire_purge_request( $post->ID );
+	}
+
+	/**
+	 * Clear on post delete.
+	 *
+	 * @param int $post_id Post ID.
+	 */
+	public function on_before_delete_post( $post_id ) {
+		$this->fire_purge_request( $post_id );
+	}
+
+	/**
+	 * Fire purge request.
+	 *
+	 * @param int    $post_id   Post ID.
+	 * @param string $permalink Permalink.
+	 */
+	protected function fire_purge_request( $post_id = 0, $permalink = '' ) {
+
+		// Get post permalink with fallback.
+		if ( empty( $permalink ) ) {
+			$permalink = get_the_permalink( $post_id );
+		}
+
+		// Get the path.
+		$path = wp_parse_url( $permalink, PHP_URL_PATH );
+
+		// Build the key.
+		$key = sprintf( '%1$s,%2$s,%3$s',
+			$path,
+			'', // Empty on purpose.
+			'site'
+		);
+
+		// Build URL.
+		$request_url = add_query_arg(
+			[
+				'endpoint' => $key,
+			],
+			home_url( '/bust-endpoint-cache' )
+		);
+
+		// Fire the request to the irving cache.
+		wp_remote_get( $request_url );
+
+		// Fire the request to WordPress VIP.
+		wp_remote_request(
+			$permalink,
+			[
+				'method' => 'PURGE',
+			]
+		);
+	}
+
+	/**
+	 * Purge cache on a PURGE request.
+	 */
+	public function purge_cache_request() {
+		global $wp;
+
+		if ( isset( $wp->request ) && 'PURGE' === strtoupper( $_SERVER['REQUEST_METHOD'] ) ) { // phpcs:ignore
+			$this->fire_purge_request( 0, home_url( $wp->request ) );
+		}
+	}
+
+	/**
+	 * Fire wipe out request.
+	 */
+	protected function fire_wipe_request() {
+		wp_remote_get( home_url( '/bust-entire-cache' ) );
+	}
+
+	/**
+	 * Render the settings page.
+	 */
+	public function register_admin() {
+		add_submenu_page(
+			'options-general.php',
+			__( 'WP-Irving Cache', 'wp-irving' ),
+			__( 'WP-Irving Cache', 'wp-irving' ),
+			'manage_options',
+			'wp-irving-cache',
+			[ $this, 'render' ]
+		);
+	}
+
+	/**
+	 * Render the settings page.
+	 */
+	public function render() {
+		// Firing request to clean cache.
+		if ( isset( $_POST['submit'] ) ) { // phpcs:ignore WordPress.Security.NonceVerification.Missing
+			$this->fire_wipe_request();
+		}
+		?>
+		<div class="wrap">
+			<h1 class="wp-heading-inline">
+				<?php esc_html_e( 'WP-Irving - Site Cache', 'wp-irving' ); ?>
+			</h1>
+
+			<hr class="wp-header-end">
+
+			<form method="post">
+				<?php settings_fields( 'irving-cache' ); ?>
+
+				<h3>
+					<?php esc_html_e( 'Clear Site Cache', 'wp-irving' ); ?>
+				</h3>
+
+				<p>
+					<?php esc_html_e( 'Use with care. Clearing the entire site cache will negatively impact performance for a short period of time.', 'wp-irving' ); ?>
+				</p>
+
+				<p class="submit">
+					<input type="submit" name="submit" id="submit" class="button" value="<?php echo esc_attr_e( 'Clear Cache', 'wp-irving' ); ?>">
+				</p>
+			</form>
+
+			<hr>
+		</div>
+		<?php
+	}
+}
+
+new \WP_Irving\Purge_Cache();
```

Comment:
Can this be instantiated outside of the class file so that including the file doesn't automatically load the purging logic? Because there is no reference to the class instance available, once it's on, it's pretty tough to turn off.